medusa/show/show.py

```diff
@@ -67,7 +67,8 @@ def find(shows, indexer_id, indexer=None):
             return None
 
         indexer_ids = [indexer_id] if not isinstance(indexer_id, list) else indexer_id
-        results = [show for show in shows if (indexer is None or show.indexer == indexer) and show.indexerid in indexer_ids]
+        results = [show for show in shows if show.imdb_id == indexer_id and indexer == 10 or
```

Comment:
Can you add a comment please? Also we should really try to avoid this excessive amount of code in list comprehensions.